File: min-cost-climbing-stairs/min-cost-climbing-stairs.scala
object Solution {
    def minCostClimbingStairs(cost: Array[Int]): Int = {
    val n = cost.length
    val dp = Array.fill(n + 1)(0)
    for (i <- 2 to n) {
      val takeOneStep = dp(i - 1) + cost(i - 1)
      val takeTwoStep = dp(i - 2) + cost(i - 2)
      dp(i) = math.min(takeOneStep, takeTwoStep)
    }
    dp(n)
  }
}

File: min-cost-climbing-stairs/min-cost-climbing-stairs.java
class Solution {
    public int minCostClimbingStairs(int[] cost) {
        int[] dp = new int[cost.length + 1];
        for (int i = 2; i <= cost.length; i++){
            int oneStep = dp[i - 1] + cost[i - 1];
            int twoStep = dp[i - 2] + cost[i - 2];
            dp[i] = Math.min(oneStep, twoStep);
        }
        return dp[cost.length];
    }
}